recipes/cyclonedds/all/conanfile.py

```diff
@@ -0,0 +1,174 @@
+        files.save(self, module_file, content)
+
+    def export_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            self.copy(patch["patch_file"])
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def requirements(self):
+        if self.options.shm:
+            self.requires("iceoryx/2.0.0")
+        if self.options.ssl:
+            self.requires("openssl/1.1.1q")
+    
+    def build_requirements(self):
+        self.tool_requires("cmake/3.16.2")
+
+    def validate(self):
+        compiler = self.settings.compiler
+        version = scm.Version(self.settings.compiler.version)
+
+        if ((self.options.security is True ) and (self.options.shared is False)):
+            raise ConanInvalidConfiguration("Cyclone DDS currently do not support"\
+                                            "static build and security on")
+
+        if compiler.get_safe("cppstd"):
+            build.check_min_cppstd(self, 14)
+
+        if compiler == "Visual Studio":
+            # ToDo : determine windows error and find solution (at test_package)
+            raise ConanInvalidConfiguration("Cyclone DDS is not (yet) supported"\
+                                                "for Visual Studio compiler.")
+        elif compiler == "gcc":
+            if version < "6":
+                raise ConanInvalidConfiguration("Using Cyclone DDS with gcc requires"\
+                                                " gcc 6 or higher.")
+            if version < "9" and compiler.get_safe("libcxx") == "libstdc++":
+                raise ConanInvalidConfiguration("gcc < 9 with libstdc++ not supported")
+            if version == "6":
+                self.output.warn("Cyclone DDS package is compiled with gcc 6, it is"\
+                                 " recommended to use 7 or higher")
+                self.output.warn("GCC 6 will build with warnings.")
+        elif compiler == "clang":
+            if compiler.get_safe("libcxx") == "libstdc++":
+                raise ConanInvalidConfiguration("clang with libstdc++ not supported")
+            if version == "7.0" and compiler.get_safe("libcxx") == "libc++" and \
+               self.options.shared and self.settings.build_type == "Debug":
+                raise ConanInvalidConfiguration("shared Debug with clang 7.0 and"\
+                                                " libc++ not supported")
+
+    def source(self):
+        files.get(self,**self.conan_data["sources"][self.version], strip_root=True,
+                 destination=self._source_subfolder)
+
+    def layout(self):
+        cmake_layout(self)
+
+    def generate(self):
+
+        tc = CMakeToolchain(self)
+        # ToDo : determine how to do in conan : 
```

Comment:
```suggestion
        # TODO : determine how to do in conan : 
```
Use always UPPER CASE for TODO, FIXE and XXX. It's much easier to filter when using grep command.